### PR TITLE
[IA-4095] Add disk types and tests

### PR DESCRIPTION
**Pull request by jdcanas**

### CODE REVIEW

File: src/libs/ajax/leonardo/models/disk-models.ts [hunk 1]
```diff
@@ -1,10 +1,87 @@
+import { AuditInfo, CloudContext, LeoResourceLabels } from 'src/libs/ajax/leonardo/models/core-models'
+
+
+export type DiskType = 'pd-standard' | 'pd-ssd' | 'pd-balanced'
+
 export interface DiskConfig {
   name: string
   size: number
   //TODO: add type when we type disks, IA-4095
```
Comment (ghost) [inline src/libs/ajax/leonardo/models/disk-models.ts]: this comment can probably be removed

File: src/libs/ajax/leonardo/models/disk-models.ts [hunk 2]
```diff
@@ -1,10 +1,87 @@
+import { AuditInfo, CloudContext, LeoResourceLabels } from 'src/libs/ajax/leonardo/models/core-models'
+
+
+export type DiskType = 'pd-standard' | 'pd-ssd' | 'pd-balanced'
```
Comment (n8landolt) [inline src/libs/ajax/leonardo/models/disk-models.ts]: Just so you're aware, Azure has different enums for their disk types.

**Review by n8landolt — APPROVED** [review]

**Review by mlilynolting — APPROVED** [review]

File: src/libs/ajax/leonardo/Disks.ts [hunk 3]
```diff
@@ -1,32 +1,34 @@
...
-  list: async (labels = {}) => {
+  list: async (labels = {}): Promise<ListDiskItem> => {
     const res = await fetchLeo(`api/google/v1/disks${qs.stringify(labels, { addQueryPrefix: true })}`,
       _.mergeAll([authOpts(), appIdentifier, { signal }]))
     return res.json()
   },
 
-  disk: (project, name) => {
+  disk: (project: string, name: string) => {
     return {
```
Comment (msilva-broad) [inline src/libs/ajax/leonardo/Disks.ts]: can this return be shorthanded to `disk: (project: string, name: string) => ({returnObject})` ?

File: src/pages/workspaces/workspace/analysis/utils/disk-utils.test.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
+import { addDays, subDays } from 'date-fns'
+import { diskStatuses } from 'src/libs/ajax/leonardo/models/disk-models'
+import {
+  generateTestDisk,
+  getRuntime, getRuntimeConfig
+} from 'src/pages/workspaces/workspace/analysis/_testData/testData'
+import { getCurrentPersistentDisk } from 'src/pages/workspaces/workspace/analysis/utils/disk-utils'
+
+
+describe('getCurrentPersistentDisk', () => {
```
Comment (msilva-broad) [inline src/pages/workspaces/workspace/analysis/utils/disk-utils.test.ts:10]: // Arrange, // Act and // Assert test section comment/headers are missing in these tests

**Pull request merged**